### PR TITLE
Add new function 'jordon'.

### PR DESCRIPTION
**Pull request by alexvong1995**
* inst/@sym/jordan.m: New file.

I use @sym/eig as a template to write this file.
The new function simply wraps around sympy's `jordan_form ()` method.

### CODE REVIEW

File: inst/@sym/jordan.m [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+%% the GNU General Public License for more details.
+%%
+%% You should have received a copy of the GNU General Public
+%% License along with this software; see the file COPYING.
+%% If not, see <http://www.gnu.org/licenses/>.
+
+%% -*- texinfo -*-
+%% @documentencoding UTF-8
+%% @deftypefn  {Function File} {@var{J} =} jordan (@var{A})
+%% @deftypefnx {Function File} {[@var{V}, @var{J}] =} jordan (@var{A})
```
Comment (cbm755) [inline inst/@sym/jordan.m]: please use `@deftypemethod  @@sym {@var{J} =} jordan (@var{A})` and `@deftypemethodx` here

File: inst/@sym/jordan.m [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+%%       ⎢0  2  0  0⎥
+%%       ⎢          ⎥
+%%       ⎢0  0  2  1⎥
+%%       ⎢          ⎥
+%%       ⎣0  0  0  2⎦
+%% @end group
+%% @end example
+%% The generalized eigenvectors are the columns of @var{V}.
+%% Those corresponding to a jordan block form a cycle.
+%% We can check those columns corresponding to the leftmost jordan block:
```
Comment (cbm755) [inline inst/@sym/jordan.m]: should `Jordan` be capitalized here?

File: inst/@sym/jordan.m [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+         '    (V, J) = sp.Matrix ([A]).jordan_form ()'};
+
+  if (nargout <= 1)
+    V = python_cmd ([cmd; {'return (J,)'}], sym (A));
+  else
+    [V, J] = python_cmd ([cmd; {'return (V, J)'}], sym (A));
+  end
+end
+
+%!test
```
Comment (cbm755) [inline inst/@sym/jordan.m]: two blank lines before tests start

File: inst/@sym/jordan.m [hunk 4]
```diff
@@ -0,0 +1,170 @@
...
+%! % matrices of unknown entries
+%! A = [sym('a') sym('b'); sym('c') sym('d')];
+%! [V, D] = eig (A);
+%! J = jordan (A);
+%! assert (isequal (simplify (D), simplify (J)));
+
+
+%%!test
+%%! % FIXME: commented out test...
```
Comment (cbm755) [inline inst/@sym/jordan.m]: Please do something like `if (python_cmd ('return Version(spver) >= Version("1.0"),'))` here.  See `@sym/mrdivide` for example.

File: inst/@sym/jordan.m (hunk 1 above)
Comment (cbm755) [inline inst/@sym/jordan.m]: see e.g., `@sym/assume.m`

File: inst/@sym/jordan.m [hunk 5]
```diff
@@ -0,0 +1,170 @@
...
+
+%% Author: Alex Vong
+%% Keywords: symbolic
+
+function [V, J] = jordan (A)
+  cmd = {'(A,) = _ins'
+         'if A.is_Matrix:'
+         '    (V, J) = A.jordan_form ()'
+         'else:'
+         '    (V, J) = sp.Matrix ([A]).jordan_form ()'};
```
Comment (cbm755) [inline inst/@sym/jordan.m]: not a big deal, but I usually follow python conventions within the python blocks: so no weird octave spaces before the `()`.

This is just FYI, up to you if you change it.

**Comment by cbm755** [comment]
Thanks, this looks good.  I'll merge after you address some minor review above.

Can you also add the function name to the list at the top of the `NEWS` file?

**Comment by alexvong1995** [comment]
Done!

**Comment by cbm755** [comment]
There is a binary "workspace" file checked in. Will need a rebase to get that out of history.

Methodx line is wrong:should like like previous line.

Lowercase Jordan still in docs.

Almost there: only the  binary file is serious.

**Comment by alexvong1995** [comment]
I only modify the commit `fix some style issues`, 
and fix all the issues within this commit.
Should we add `octave-workplace` to `.gitiignore`?
So that it makes it harder to commit it accidentally.

**Comment by cbm755** [comment]
Merged, thanks!

FYI: please see 511ef26abbb63e63c13236f437d4d2119ffc8f9d for the correct `deftypemethod` texinfo.

re: .gitignore: great idea, done in e5d3aef600ebd78200d6583ac8637f969fe35f7a

**Pull request merged**